src/coreclr/vm/jitinterface.h

```diff
@@ -103,6 +103,10 @@ BOOL LoadDynamicInfoEntry(Module *currentModule,
 
 #endif // TARGET_X86
 
+// These must be implemented by the JIT in order to support the "jithelpers" functionality.
+EXTERN_C void JIT_Patchpoint(int* counter, int ilOffset);
```

Comment:
```suggestion
EXTERN_C FCDECL2(void, JIT_Patchpoint, int* counter, int ilOffset);
```

Reply:
This is an existing oddity, and your suggested breaks the build on Windows X86. Given that this change is already very much oversized, I don't want to change this.

Reply:
> your suggested breaks the build on Windows X86

Right, there is a trivial matching change required in .cpp file that github won't allow me to leave as a suggestion:

```
void JIT_Patchpoint(int* counter, int ilOffset)`
```
->
```
HCIMPL2(void, JIT_Patchpoint, int* counter, int ilOffset)
...
HCIMPLEND
```

Reply:
This PR is moving `JIT_Patchpoint`'s implementation to assembly, is there a way to represent HCIMPL2's expansion in asm?

Reply:
@am11, no there isn't any way to do that which I am aware of. An alternative approach would have been to add a new JIT intrinsic which can get the return address, and then implement this via an FCALL/QCALL pair, but I judged that wasn't actually simpler to do, and I didn't want to have a single use JIT intrinsic here.